Fix some raiders missing swimspeed modifiers

someaddons opened the pull request:
# Changes proposed in this pull request:
Fix some raiders missing swimspeed modifiers and fix all their values
Fix some of our entities not using our custom movement controller
Fix our entities beeing affected by bubble columns

Review please

someaddons committed:
Fix some raiders missing swimspeed modifiers

someaddons committed:
Fix some raiders missing swimspeed modifiers

ravenbuilder934 commented on src/api/java/com/minecolonies/api/entity/mobs/vikings/AbstractEntityNorsemen.java:
```diff
@@ -20,6 +20,11 @@
+    /**
+     * Swim speed for pirates
+     */
+    private static final double PIRATE_SWIM_BONUS = 1.5;
```
are they pirates though?

someaddons replied:
sure^^

Raycoms commented on src/api/java/com/minecolonies/api/entity/mobs/AbstractEntityMinecoloniesMob.java:
```diff
@@ -235,6 +235,8 @@ public AbstractAdvancedPathNavigate getNavigation()
         return newNavigator;
     }
 
+    public abstract double getSwimSpeedFactor();
```
no javadoc =O

someaddons committed:
javadoc

someaddons committed:
javadoc

Raycoms committed:
structurize bump

ravenbuilder934 committed:
Schem fixes (#7381)

* Fix some broken schems

someaddons committed:
Adjust swimspeed modifiers further
Make citizens no longer stuck in flowing water
All our entities now use our custom movecontroller

someaddons committed:
Adjust swimspeed modifiers further
Make citizens no longer stuck in flowing water
All our entities now use our custom movecontroller

Raycoms committed:
Fix citizen armor updates - 6905 (#7374)

Fix issue of guards switching gear mid combat because another piece of gear is earlier in the inventory.
Adjust the armor updates to be more efficient and less convoluted.

someaddons committed:
Fix integrated buildings possible to be picked up (#7376)

Make integrated buildings impossible to be picked up
make builder skip clear step on integrated schematics

Raycoms committed:
Do LevenshteinDistance setting. (#7377)

Use a propper sorting algorithm to allow players to find things better.

someaddons committed:
Monster list is now immutable instead of copied (#7378)

Monster list is now immutable instead of copied
Citizensounds only get played when a player is close enough

Raycoms commented on src/api/java/com/minecolonies/api/entity/citizen/AbstractEntityCitizen.java:
```diff
@@ -734,4 +740,24 @@ public void updatePose(final Pose pose)
+     *
+     * @param down
+     */
+    public void onInsideBubbleColumn(boolean down)
```
Maybe we want to allow upward movement and not downward?

someaddons replied:
they do swim upwards by themselves, they just no longer will be hindered in where they want to move by the column itself

Raycoms committed:
Finish refactor (#7383)

This PR finally refactors the townhall UI. It seperates each of the pages into its own Window with the following benefits:
a) Better separation of concerns, it's easier to extend existing pages and easier to find certain functionalities.
b) It cleans up a lot of the hacky page management resulting from the double navigation (tabs + pages).

uecasm committed:
Bump JEI and cache recipes to improve startup perf (#7382)

Bump JEI to slightly newer build and use newer methods for better translation support
Generate generic vanilla recipes once in advance instead of per module to improve startup speed

Raycoms commented:
Just missing the conflicts then :shipit:

someaddons committed:
bump structurize

someaddons committed:
Merge remote-tracking branch 'remotes/origin/swimspeed' into swimspeed

# Conflicts:
#	src/api/java/com/minecolonies/api/entity/citizen/AbstractEntityCitizen.java
#	src/main/resources/META-INF/accesstransformer.cfg
